sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/PostgresConnectionProvider.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.jdbc.connection
+
+import java.sql.{Connection, Driver}
+import javax.security.auth.login.{AppConfigurationEntry, Configuration}
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions
+import org.apache.spark.sql.execution.datasources.jdbc.connection.PostgresConnectionProvider.PGJDBCConfiguration
+import org.apache.spark.util.SecurityUtils
+
+private[jdbc] class PostgresConnectionProvider(driver: Driver, options: JDBCOptions)
+    extends BasicConnectionProvider(driver, options) {
+  def setAuthenticationConfigIfNeeded(): Unit = {
+    val parent = Configuration.getConfiguration
+    val configEntry = parent.getAppConfigurationEntry(PostgresConnectionProvider.appEntry)
+    if (configEntry == null || (configEntry != null && configEntry.isEmpty)) {
```

Comment:
`if (configEntry == null || configEntry.isEmpty)` will do the same as "short circuit"

Reply:
Yep, I was oversafe so simplified.